### PR TITLE
refactor store heartbeat

### PR DESCRIPTION
**Pull request by follitude**
This PR is separated from #2120. 
PTAL @zhangjinpeng1987  @siddontang

### CODE REVIEW

File: src/raftstore/store/store.rs [hunk 1]
```diff
@@ -98,6 +96,14 @@ impl Default for StoreStat {
     }
 }
 
+pub struct StoreInfo {
+    pub engine: Arc<DB>,
+    pub capacity: u64,
+    pub tag: String,
```
Comment (siddontang) [inline src/raftstore/store/store.rs]: seem you don't need to pss tag, start time and is_busy

Comment (follitude) [inline src/raftstore/store/store.rs]: ok

File: src/raftstore/store/worker/pd.rs [hunk 2]
```diff
@@ -157,7 +165,49 @@ impl<T: PdClient> Runner<T> {
...
+        let disk_cap = disk_stats.total_space();
+        let capacity = if store_info.capacity == 0 || disk_cap < store_info.capacity {
+            disk_cap
+        } else {
+            store_info.capacity
+        };
+        stats.set_capacity(capacity);
+
+        stats.used_size += flush_engine_properties_and_get_used_size(store_info.engine.clone());
+
```
Comment (zhangjinpeng1987) [inline src/raftstore/store/worker/pd.rs]: Please split this function into two independent functions.

Comment (zhangjinpeng1987) [inline src/raftstore/store/worker/pd.rs]: stats.used_size += ??? Is this works?

File: src/raftstore/store/worker/pd.rs [hunk 3]
```diff
@@ -459,3 +523,23 @@ fn send_destroy_peer_message(
         )
     }
 }
+
+pub fn get_engine_used_size(engine: Arc<DB>) -> u64 {
```
Comment (zhangjinpeng1987) [inline src/raftstore/store/worker/pd.rs]: This function not belongs to here.

File: src/raftstore/store/store.rs [hunk 4]
```diff
@@ -1895,13 +1853,13 @@ impl<T: Transport, C: PdClient> Store<T, C> {
             }
         }
 
+        stats.set_start_time(self.start_time.sec as u32);
```
Comment (zhangjinpeng1987) [inline src/raftstore/store/store.rs]: Move this line to 1862.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**